x/staking/keeper/delegation.go

```diff
@@ -517,7 +517,9 @@ func (k Keeper) unbond(ctx sdk.Context, delAddr sdk.AccAddress, valAddr sdk.ValA
 
 	// if the delegation is the operator of the validator and undelegating will decrease the validator's self delegation below their minimum
 	// trigger a jail validator
-	if isValidatorOperator && !validator.Jailed && validator.DelegatorShareExRate().Mul(delegation.Shares).TruncateInt().LT(validator.MinSelfDelegation) {
+	if isValidatorOperator && !validator.Jailed &&
+		validator.ShareTokens(delegation.Shares).TruncateInt().LT(validator.MinSelfDelegation) {
```

Comment:
Should we use `ShareTokensTruncated` here (in addition to `TruncateInt`)?

Reply:
These do two fundamentally different things... Truncation _to_ an `Int` vs truncation of the final decimal place

Reply:
I realize this - I am wondering if we should check the self-bond using the `ShareTokensTruncated` version of the calculation.

Reply:
Oh "in addition" didn't register when I first read your comment. 

I think it makes sense as is, given that the token worth is considered to be rounded (aka not truncated) token amount derived from delegation shares (as per `validator.ShareTokens()`)